Composer/packages/client/src/recoilModel/dispatchers/provisionQNA.ts

```diff
@@ -0,0 +1,220 @@
+          .primaryKey;
+        const appInsightsComponent = await applicationInsightsManagementClient.components.get(
+          resourceGroupName,
+          resourceGroupName
+        );
+        const userAppInsightsKey = appInsightsComponent.instrumentationKey;
+        const userAppInsightsName = resourceGroupName;
+        const userAppInsightsAppId = appInsightsComponent.appId;
+        const primaryEndpointKey = `${qnaMakerWebAppName}-PrimaryEndpointKey`;
+        const secondaryEndpointKey = `${qnaMakerWebAppName}-SecondaryEndpointKey`;
+        const defaultAnswer = 'No good match found in KB.';
+        const QNAMAKER_EXTENSION_VERSION = 'latest';
+
+        // deploy qna host webapp
+        const webAppResult = await webSiteManagementClient.webApps.createOrUpdate(
+          resourceGroupName,
+          qnaMakerWebAppName,
+          {
+            name: qnaMakerWebAppName,
+            serverFarmId: resourceGroupName,
+            location: region,
+            siteConfig: {
+              cors: {
+                allowedOrigins: ['*'],
+              },
+              appSettings: [
+                {
+                  name: 'AzureSearchName',
+                  value: qnaMakerSearchName,
+                },
+                {
+                  name: 'AzureSearchAdminKey',
+                  value: azureSearchAdminKey,
+                },
+                {
+                  name: 'UserAppInsightsKey',
+                  value: userAppInsightsKey,
+                },
+                {
+                  name: 'UserAppInsightsName',
+                  value: userAppInsightsName,
+                },
+                {
+                  name: 'UserAppInsightsAppId',
+                  value: userAppInsightsAppId,
+                },
+                {
+                  name: 'PrimaryEndpointKey',
+                  value: primaryEndpointKey,
+                },
+                {
+                  name: 'SecondaryEndpointKey',
+                  value: secondaryEndpointKey,
+                },
+                {
+                  name: 'DefaultAnswer',
+                  value: defaultAnswer,
+                },
+                {
+                  name: 'QNAMAKER_EXTENSION_VERSION',
+                  value: QNAMAKER_EXTENSION_VERSION,
+                },
+              ],
+            },
+            enabled: true,
+          }
+        );
+
+        // Create qna account
+        const cognitiveServicesManagementClient = new CognitiveServicesManagementClient(
+          tokenCredentials,
+          subscriptionId
+        );
+        await cognitiveServicesManagementClient.accounts.create(resourceGroupName, qnaMakerServiceName, {
+          kind: 'QnAMaker',
+          sku: {
+            name: 'F0',
+          },
+          location: region,
+          properties: {
+            apiProperties: {
+              qnaRuntimeEndpoint: `https://${webAppResult.hostNames?.[0]}`,
```

Comment:
does it make sense wrap the whole account creation inside if (webAppResult.hostNames?\[0\]) as the the endpoint has a chance of https://undefined here.

Reply:
This should never happen because a failure of the webapp call will throw an error, HOWEVER I did a test for this condition and threw my own error just in case.